internal/search/query/predicate.go

```diff
@@ -81,36 +81,52 @@ func (f *RepoContainsPredicate) ParseParams(params string) error {
 	}
 
 	for _, node := range nodes {
-		switch v := node.(type) {
-		case Parameter:
-			switch strings.ToLower(v.Field) {
-			case "file":
-				if f.File != "" {
-					return errors.New("cannot specify file multiple times")
-				}
-				f.File = v.Value
-			case "content":
-				if f.Content != "" {
-					return errors.New("cannot specify content multiple times")
-				}
-				f.Content = v.Value
-			default:
-				return fmt.Errorf("unsupported option %q", v.Field)
+		if err := f.parseNode(node); err != nil {
+			return err
+		}
+	}
+
+	if f.File == "" && f.Content == "" {
+		return errors.New("one of file or content must be set")
+	}
+
+	return nil
+}
+
+func (f *RepoContainsPredicate) parseNode(n Node) error {
+	switch v := n.(type) {
+	case Parameter:
+		switch strings.ToLower(v.Field) {
+		case "file":
+			if f.File != "" {
+				return errors.New("cannot specify file multiple times")
 			}
-		case Pattern:
+			f.File = v.Value
+		case "content":
 			if f.Content != "" {
 				return errors.New("cannot specify content multiple times")
 			}
 			f.Content = v.Value
 		default:
-			return fmt.Errorf("unsupported node type %T", node)
+			return fmt.Errorf("unsupported option %q", v.Field)
 		}
+	case Pattern:
+		if f.Content != "" {
+			return errors.New("cannot specify content multiple times")
+		}
+		f.Content = v.Value
+	case Operator:
+		if v.Kind == Or {
+			return errors.New("predicates do not currently support or queries")
```

Comment:
```suggestion
			return errors.New("predicates do not currently support 'or' queries")
```

We don't support `not` either right? Can we add a cases for that (check `v.Negated` for Parameter, Content, and `v.Kind == Not`).